exporter/awsemfexporter/config.go

```diff
@@ -66,3 +73,51 @@ type Config struct {
 	// logger is the Logger used for writing error/warning logs
 	logger *zap.Logger
 }
+
+type MetricDescriptor struct {
+	// metricName is the name of the metric
+	metricName string `mapstructure:"metric_name"`
+	// unit defines the overwritten value of metric descriptor `unit`
+	unit string `mapstructure:"unit"`
+	// overwrite set to true means the metric descriptor will be overwritten if it is set before emf exporter reads the metric; false means
+	// the descriptor will only by configured if empty.
+	overwrite bool `mapstructure:"overwrite"`
+}
+
+func (config *Config) Validate() {
+	validDeclarations := []*MetricDeclaration{}
+	for _, declaration := range config.MetricDeclarations {
+		err := declaration.Init(config.logger)
+		if err != nil {
+			config.logger.Warn("Dropped metric declaration. Error: " + err.Error() + ".")
```

Comment:
Use the `WithFields` (or WithError if available in this logger) instead of building this string.

Reply:
Fixed